packages/@tailwindcss-postcss/src/index.ts

```diff
@@ -62,27 +64,54 @@ function tailwindcss(opts: PluginOptions = {}): AcceptedPlugin {
       {
         postcssPlugin: 'tailwindcss',
         async Once(root, { result }) {
-          env.DEBUG && console.time('[@tailwindcss/postcss] Total time in @tailwindcss/postcss')
+          using I = new Instrumentation()
+
           let inputFile = result.opts.from ?? ''
+          DEBUG && I.start(`[@tailwindcss/postcss] ${relative(base, inputFile)}`)
+
+          DEBUG && I.start('Quick bail check')
+          let canBail = true
+          root.walkAtRules((node) => {
+            if (
+              node.name === 'import' ||
+              node.name === 'theme' ||
+              node.name === 'config' ||
+              node.name === 'plugin' ||
+              node.name === 'apply'
+            ) {
+              canBail = false
+              return false
+            }
+          })
+          if (canBail) return
```

Comment:
This is the quick bail where we are sure that Tailwind CSS should not be used. This prevents creating the compiler entirely.

Reply:
Should this be in a separate commit?

Reply:
I think you are right, it started this way and then diverged into 2 PRs. Created a separate one here: https://github.com/tailwindlabs/tailwindcss/pull/15340